File: web/fix-headers.js
const fs = require('fs');
const glob = require('glob'); // npm package or just simple recursiv readdir

function walk(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(function (file) {
        file = dir + '/' + file;
        const stat = fs.statSync(file);
        if (stat && stat.isDirectory()) {
            results = results.concat(walk(file));
        } else {
            if (file.endsWith('page.js') || file.endsWith('ArchiveBrowser.js')) results.push(file);
        }
    });
    return results;
}

const files = walk('c:/Users/hoon/Desktop/els_home_v1/web/app/employees');

files.forEach(file => {
    let content = fs.readFileSync(file, 'utf8');
    if (content.includes('className={styles.headerBanner}') && content.includes('className={styles.controls}')) {
        let lines = content.split('\n');
        let bannerIdx = -1, bannerEnd = -1, controlsStart = -1, controlsEnd = -1;

        for (let i = 0; i < lines.length; i++) {
            if (lines[i].includes('<div className={styles.headerBanner}>')) bannerIdx = i;
            if (bannerIdx !== -1 && lines[i].includes('</h1')) bannerEnd = i + 1;
            if (lines[i].includes('className={styles.controls}')) controlsStart = i;
        }

        if (bannerIdx !== -1 && controlsStart !== -1 && controlsStart === bannerEnd + 1) {
            let divCount = 0;
            for (let i = controlsStart; i < lines.length; i++) {
                divCount += (lines[i].match(/<div/g) || []).length;
                divCount -= (lines[i].match(/<\/div/g) || []).length;
                if (divCount === 0) {
                    controlsEnd = i;
                    break;
                }
            }
            if (controlsEnd !== -1) {
                // remove the </div> at bannerEnd
                lines.splice(bannerEnd, 1);
                // insert </div> at controlsEnd
                lines.splice(controlsEnd, 0, '            </div>');
                fs.writeFileSync(file, lines.join('\n'));
                console.log('Fixed', file);
            }
        }
    }
});

File: web/fix-headers.py
import os
import re

directory = "c:/Users/hoon/Desktop/els_home_v1/web/app/employees"

pattern = re.compile(r'(<div className=\{styles\.headerBanner\}>\s*<h1 className=\{styles\.title\}>[^<]*</h1>\s*)</div>\s*(<div className=\{styles\.controls\}[^>]*>.*?(?:</div>))', re.DOTALL)

for root, _, files in os.walk(directory):
    for file in files:
        if file.endswith("page.js") or file.endswith("ArchiveBrowser.js"):
            filepath = os.path.join(root, file)
            with open(filepath, 'r', encoding='utf-8') as f:
                content = f.read()

            def repl(m):
                # We need to correctly capture the end of the controls div if it has nested children.
                # Actually, an easier regex is replacing </div> followed by <div className={styles.controls} ...> 
                return m.group(1) + m.group(2) + '\n            </div>'
            
            # Since the regex .*? </div> might be greedy or not greedy enough (if the controls has nested divs), 
            # I should use a simpler string replace instead or just use the multi_replace_file_content tool, but there are so many files.
            pass
